src/sentry/api/serializers/rest_framework/sentry_app.py

```diff
@@ -34,3 +34,20 @@ class SentryAppSerializer(Serializer):
     redirectUrl = serializers.URLField(required=False)
     isAlertable = serializers.BooleanField(required=False)
     overview = serializers.CharField(required=False)
+
+    def validate_events(self, attrs, source):
+        if not attrs.get('scopes'):
+            return attrs
+        resources = [s.split(':')[0] for s in attrs['scopes']]
+        for event in attrs[source]:
+            if event == 'issue':
+                resource = 'event'
+            else:
+                resource = event
+
+            if resource not in resources:
+                raise ValidationError(
+                    u"resource type '{}' does not have the correct permissions.".format(event),
+                )
```

Comment:
If you choose "Issue: Admin" permissions, you get `event:read` and `event:write` in addition to `event:admin`.

I think you could simplify this code a bit by just looking for `read` specifically. It also leaves the door open for anything down the line that requires a `write` or above specifically.

```python
for event in attrs[source]:
    if event == 'issue':
        scope_needed = 'event:read'

    if scope_needed not in attrs['scopes']:
        raise ValidationError(...)
```